frontend/src/components/accounts/Accounts.tsx

```diff
@@ -2,44 +2,45 @@ import React from "react";
 
 import AccountsApi, * as A from "../../libraries/explorer-wamp/accounts";
 
-import AccountsList from "./AccountsList";
+import autoRefreshHandler from "../utils/autoRefreshHandler";
 import PaginationSpinner from "../utils/PaginationSpinner";
+import FlipMove from "../utils/FlipMove";
+import AccountsList from "./AccountsList";
 
-export interface Props {}
-
-export interface State {
-  accounts: A.AccountBasicInfo[] | null;
+export interface OuterProps {
+  count: number;
 }
 
-export default class extends React.Component<Props, State> {
-  state: State = {
-    accounts: null
+export default class extends React.Component<OuterProps> {
+  static defaultProps = {
+    count: 15
   };
 
-  _accountApi: AccountsApi | null;
-
-  constructor(props: Props) {
-    super(props);
-    this._accountApi = null;
-  }
-
   fetchAccounts = async () => {
-    if (this._accountApi === null) {
-      this._accountApi = new AccountsApi();
-    }
-    const accounts = await this._accountApi.getAccounts();
-    this.setState({ accounts });
+    return await new AccountsApi().getAccounts(this.props.count);
   };
 
-  componentDidMount() {
-    this.fetchAccounts();
+  autoRefreshAccounts = autoRefreshHandler(Accounts, this.fetchAccounts);
+
+  render() {
+    return <this.autoRefreshAccounts />;
   }
+}
 
+interface InnerProps {
+  items: A.AccountBasicInfo[];
+}
+class Accounts extends React.Component<InnerProps> {
   render() {
-    const { accounts } = this.state;
-    if (accounts === null) {
-      return <PaginationSpinner hidden={false} />;
+    const { items } = this.props;
+    let accountsShow = <PaginationSpinner hidden={false} />;
+    if (items.length > 0) {
+      accountsShow = (
+        <FlipMove duration={1000} staggerDurationBy={0}>
+          <AccountsList accounts={items} />
+        </FlipMove>
+      );
     }
-    return <AccountsList accounts={accounts} />;
+    return <>{accountsShow}</>;
```

Comment:
I like the style we handle spinner in the Nodes.tsx better:

```tsx
const { items } = this.props;
if (items.length === 0) {
  return <PaginationSpinner hidden={false} />;
}
return (
  <FlipMove duration={1000} staggerDurationBy={0}>
    <AccountsList accounts={items} />
  </FlipMove>
);
```

Let's update all the components to use this style.